### ISSUE TITLE
Some MPs are incorrectly connected to speeches that conflict with the chamber they belonged to

### ISSUE THREAD

**Issue by fredrik1984** [swerik-project/riksdagen-records#47]
For example, in this record from the second chamber – https://github.com/swerik-project/riksdagen-records/blob/main/data/1926/prot-1926--ak--044.xml – the speech introduction "Hans excellens herr statsministern Sandler:" is connected to Karl Schlyter (1879–1959) (i-NdmzAJFCHr1sfnsvdVwR4y) who belonged to the first chamber.

Hence, we should create a test that makes sure that these kinds of conflicts are detected. That is, an MP that belongs to a certain chamber (and lived in a certain period), should not be connected to a speech that is found in another chamber (and in another time period).

**Comment by fredrik1984** [comment]
Yet, Sandler was prime minister, hence he was allowed to speak in both chambers. However, it should be possible to create another test to check this, and to simply improve the mapping algorithm such as focusing on all these "Hans excellens" introductions of ministers

**Comment by ninpnin** [comment]
I think we had a discussion a long time ago that we don't know if people from one chamber can speak in the other one. Could we get an expert opinion from the library maybe?

**Comment by MansMeg** [comment]
I agree we should try to get a unittest for this somehow. Although, it is not clear how and what to test.

**Comment by ninpnin** [comment]
I remember us writing the mapping algorithm _specifically_ so that we allow people from the other chamber to be tagged, in case nobody in the same chamber was matched.

**Comment by fredrik1984** [comment]
I can check this, but I am pretty sure ministers are allowed to speak in both chambers.

Still, I think that this can also be fixed be fine-tuning the mapping algorithm to take more complex speaker introductions into account, like "herr excellens", "friherre", "grefve" etc

**Comment by fredrik1984** [comment]
I just go a response from Lotta about this, who in turn got a response from Joakim Scherp (the inhouse historian at the parliament) who wrote the following below. I have marked the quote from the Riksdag Act 1866 (§ 53, see attached image) in bald that states that ministers are allowed to _speak_ in the chamber but not take part in the decision/voting (no matter if they are MPs or not, and no matter which chamber they belong to). 

Det verkar inte som om statsråd under tvåkammartiden var helt uteslutna från att vara medlemmar av kammaren.
 
Enligt RO 1866 kunde de dock inte delta i utskott eller val till utskott (§ 43), men beroende på hur man tolkar § 53 kan de ha haft rätt att votera i kamrarma:
”**Uti hvardera Kammaren ega Statsrådets Ledamöter tillträde, med rättighet att deltaga i öfverläggningarna, men ej i besluten**, der de ej äro Ledamöter af Kammaren.”
Antingen betyder detta att statsråden inte kan delta i voteringarna eftersom de inte är ledamöter av kamrarna, eller, mer troligt, betyder det att de inte kan delta för så vitt de inte är ledamöter, för då kan de delta.
 
Men helt klart är att statsråden inte automatiskt var ledamöter av riksdagen. Detta var ju i enlighet med den maktdelningslära som var rådande under 1800-talet: kungen och statsråden utgjorde en statsmakt separerad från den andra statsmakten, som var riksdagen. Till skillnad från kungen hade dock statsråden rätt att närvara i riksdagen och delta i dess debatter oavsett om de var medlemmar av riksdagen eller ej (som ju framgår av citatet).


![Bokutdrag](https://github.com/user-attachments/assets/cd1bc20c-c169-48dd-ad9a-8c653657b73a)
